### PR TITLE
[WIP] Add Facebook login integration

### PR DESCRIPTION
**Pull request by lucianomlima**
I will need some help with Flow, Relay and GraphQL in this PR.

### CODE REVIEW

File: src/utils/facebook.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { AccessToken, GraphRequest, GraphRequestManager, LoginManager } from 'react-native-fbsdk';
+
+type GraphRequestConfig<O, S> = {
+  accessToken: O,
+  version: S,
+};
+
+const permissions: string[] = ['public_profile', 'email'];
+
+const graphRequest = (path: string = '/me', accessToken: {}, callback: () => void) => {
```
Comment (lXSPandora) [inline src/utils/facebook.js]: @lucianomlima To type objects you can use like this `acessToken: Object`

File: src/utils/facebook.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  const config: GraphRequestConfig<{}, string> = {
+    accessToken,
+    version: 'v3.0',
+  };
+  const infoRequest = new GraphRequest(path, config, callback);
+  new GraphRequestManager().addRequest(infoRequest).start();
+};
+
+const loginGraphRequest = (accessToken: {}, callback: () => void) => {
```
Comment (lXSPandora) [inline src/utils/facebook.js]: same

File: src/utils/facebook.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            }
+          },
+          () => {
+            reject(new Error('Login failed'));
+          },
+        );
+      });
+  });
+
+export default {
```
Comment (lXSPandora) [inline src/utils/facebook.js]: i think its better to not use export default use export for each function its better for the js bundle <3

**Comment by jgcmarins** [comment]
@lucianomlima what is missing on this PR?

**Comment by lucianomlima** [comment]
@jgcmarins Need some help with the relay step. In server, needs to add a mutation for the facebook login. I don't have relay/graphql experience.

**Comment by jgcmarins** [comment]
we can start by creating an issue with what is missing

**Pull request closed without merging**